### ISSUE TITLE
New encryption that il2cppdumper cannot decrypt

### ISSUE THREAD

**Issue by edsonmods** [Perfare/Il2CppDumper#489]
When I'm going to use il2cppdumper, it shows the error below...
![ERRO METADATA](https://user-images.githubusercontent.com/88516804/128449900-59946fe2-ec42-4f7a-9ee8-78e49c507a2f.png)

from what i tested, the free fire game encrypted the global-metadata.dat file, so i looked for the global-metadata.dat in the .BIN files, searching for the value "AF 1B B1 FA" in all the .BIN files dumped by the game guardian, but I didn't find the value...

so I can dump with il2cppdumper, I need to decrypt the global-metadata.dat, does anyone know another way to get the global-metadata.dat? I already tried searching for "AF 1B B1 FA" in all .BIN files dumped by game guardian but I didn't find the value in any of the .BIN files.

I also used the other project "Riru-Il2CppDumper" with magisk but I couldn't decrypt the dump...
I also used bryan's libdumper, but it doesn't download the global-metadata.dat file because it's encrypted...
I also deleted the pre-ELF data in the BIN file...

in summary, i need to decrypt the global-metadata.dat or find another way to find it to get the dump.

->files:
8A709000
![Maps txt ](https://user-images.githubusercontent.com/88516804/128451636-c43b12bc-1170-4202-9f21-09ca53cf7789.png)

->Bin correct
https://www.mediafire.com/file/bobh3bbb0mdhqw5/ARCHIVE_BIN_DECRYPT.bin/file

->global-metadata.dat
https://www.mediafire.com/file/oaenhgjxh7kbhnn/global-metadata.dat/file

-> GAME:
GARENA FREE FIRE
-> VERSION
1.64

**Comment by muathudaudon** [comment]
No one can solve

**Comment by HighMod** [comment]
Riru il2cppdumper has been recently updated. Can you try again? I don't have root to try sadly

**Comment by muathudaudon** [comment]
> Riru il2cppdumper has been recently updated. Can you try again? I don't have root to try sadly

I tried but not work, maybe frida can

**Comment by Perfare** [comment]
https://github.com/Perfare/Il2CppDumper#error-metadata-file-supplied-is-not-valid-metadata-file

**Comment by IRCKALUS** [comment]
https://www.mediafire.com/file/51olrdua6lihjd3/dump_byIRCKALUS.rar/file

dump full free fire 1.64.X enjoy :) @ircready my telegram
